rmw_fastrtps_shared_cpp/test/test_rmw_init_options.cpp

```diff
@@ -178,6 +178,38 @@ TEST(RMWInitOptionsTest, copy) {
   EXPECT_STREQ("/root", options.security_options.security_root_path);
 }
 
+static void * failing_allocate(size_t size, void * state)
```

Comment:
You definitely don't need to change this here, but one thing I learned while writing failing_allocators like this, is that you can just define the function without parameter names to avoid unused variable errors.

```
static void * failing_allocate(size_t , void* ) { return NULL; }
```

Reply:
I've learned not to do that because of compiler warnings. My memory may be failing me here though.

Reply:
Oh funny! I sometimes see compiler warnings when cast to void.